stdlib/fun.mli

```diff
@@ -61,3 +63,67 @@ exception Finally_raised of exn
     an unexpected exception or a programming error. As a general rule,
     one should not catch a [Finally_raised] exception except as part of
     a catch-all handler. *)
+
+(** {1:examples Examples}
+
+    {2 Combinators}
+
+    One common mode of usage for {!combinators} is to create and pass anonymous
+    functions as arguments to higher-order functions, without needing to specify
+    [(fun ... -> ...)] explicitly.
+
+    The examples below will be demonstrating this mainly with the {!module:List}
+    module.
```

Comment:
I am wondering if `Seq` would not be a better illustration. In particular, we could have a section on building sequences to illustrate `const` and `id` and `Seq` has more sequence constructor like `forever` or `ints` to use as a basis of more examples.

Reply:
The problem with using `Seq`, answering also your comment on `const` above, is in three aspects:

- it's less straightforward due to its delay semantics, and its potentially infinite nature.. Instead of focusing on one function, there's a potential to add more distractions (`List.of_seq`, `Seq.take`, `Seq.equal`...)
- it's less common in code and pedagogical material. list manipulation is one of the first things beginners work through
- it doesn't play as nicely with the toplevel evaluation theme we have going on, because `Seq.t` prints as `<fun>`